### PR TITLE
Create image

### PR DESCRIPTION
**Pull request by AMBALERT1**
- [x] - created a migration and model for images 
- [x] - added association to the model for images 
- [x] - Create an endpoint that is going to handle image creation for a specific spot ID.

### CODE REVIEW

File: backend/routes/api/spots.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const { user } = req;
+    const { address , city , state, country , lat, lng, name, description, price } = req.body;
+    const validationErrors = validationResult(req);
+
+    if(!validationErrors.isEmpty()) {
+        return res.status(400).json({ errors: validationErrors.array() });
+    }
+
+    const newSpot = await Spot.create({
+        owneerId: user.id,
```
Comment (AnthonyBronca) [inline backend/routes/api/spots.js]: I think you have a typo

**Review by AnthonyBronca — CHANGES_REQUESTED** [review]

**Review by AnthonyBronca — COMMENTED** [review]
Looks good, but may have a typo in one of the files

File: backend/routes/api/spots.js (hunk 1 above)
Comment (AMBALERT1) [reply backend/routes/api/spots.js]: I have corrected the typo

**Comment by AMBALERT1** [comment]
changed the typo

File: backend/routes/api/spots.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  check('lng').isFloat({ min: -180, max: 180 }).withMessage('Longitude must be between -180 and 180.'),
+  check('name').exists({ checkFalsy: true }).withMessage('Name is required.'),
+  check('description').exists({ checkFalsy: true }).withMessage('Description is required.'),
+  check('price').isFloat({ gt: 0 }).withMessage('Price must be greater than 0.'),
+];
+
+router.post('/', requireAuth, validateSpot, async (req,res) => {
+    const { user } = req;
+    const { address , city , state, country , lat, lng, name, description, price } = req.body;
+    const validationErrors = validationResult(req);
```
Comment (AnthonyBronca) [inline backend/routes/api/spots.js]: Why are you using validationResult?

**Review by AnthonyBronca — CHANGES_REQUESTED** [review]

File: backend/routes/api/spots.js (hunk 2 above)
Comment (AMBALERT1) [reply backend/routes/api/spots.js]: I was using ValidationResult from the express - validator to be able to handle and check the results of those validations checks that I defined in my validateSpot middleware

Comment (AnthonyBronca) [reply backend/routes/api/spots.js]: Can you provide the docs you referenced for this, and explain why we may not want to use this function?

Comment (AMBALERT1) [reply backend/routes/api/spots.js]: https://docs.greatexpectations.io/docs/reference/learn/terms/validation_result/#:~:text=A%20Validation%20Result%20is%20an,Validates

**Comment by AMBALERT1** [comment]
Updated the code in regards to validation result

**Review by AnthonyBronca — COMMENTED** [review]
Looks good

**Pull request merged**